compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/virtual/CommitAllocationNode.java

```diff
@@ -141,7 +141,8 @@ public void virtualize(VirtualizerTool tool) {
         for (int i = 0; i < virtualObjects.size(); i++) {
             VirtualObjectNode virtualObject = virtualObjects.get(i);
             int entryCount = virtualObject.entryCount();
-            tool.createVirtualObject(virtualObject, values.subList(pos, pos + entryCount).toArray(new ValueNode[entryCount]), getLocks(i), ensureVirtual.get(i));
+            /* n.b. the node source position of virtualObject will have been set when it was created */
+            tool.createVirtualObject(virtualObject, values.subList(pos, pos + entryCount).toArray(new ValueNode[entryCount]), getLocks(i), null, ensureVirtual.get(i));
```

Comment:
You could pass `virtualObject.getNodeSourcePosition()` instead of null to indicate where you except the value to come from.